plugins/woocommerce-admin/client/products/fields/category-field/category-field-item.tsx

```diff
@@ -89,7 +87,7 @@ export const CategoryFieldItem: React.FC< CategoryFieldItemProps > = ( {
 				<CheckboxControl
 					label={ item.data.name }
 					checked={ selectedIds.includes( item.data.id ) }
-					onChange={ () => item.data /*&& onSelect( item.data )*/ }
```

Comment:
Don't need the `onSelect` anymore as we are making using of the `getItemProps` props that handles this.